### ISSUE TITLE
False positive -Wreturn-type for bitfields

### ISSUE THREAD

**Issue by marxin** [llvm/llvm-project#52666]
The following test-case is wrongly marked with `-Wreturn-type`:

```c++
struct A
{
  unsigned int val: 1;
};

int foo(struct A *a)
{
  switch (a->val)
  {
    case 0:
      return 123;
    case 1:
      return 456;
  }
}
```

```
$ clang unreachable.c  -Werror=return-type -c
unreachable.c:15:1: error: non-void function does not return a value in all control paths [-Werror,-Wreturn-type]
}
^
1 error generated.
```

Note GCC accepts the code:
```
gcc unreachable.c  -Werror=return-type -c
```

**Comment by AaronBallman** [comment]
Resolving as a duplicate of #111509
